src/main/java/sg/ncl/MainController.java

```diff
@@ -3494,6 +3486,514 @@ public String adminEnergy(Model model,
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        if (id.isPresent()) {
+            HttpEntity<String> request = createHttpEntityHeaderOnly();
+            ResponseEntity response = restTemplate.exchange(properties.getMonthly() + "/" + id.get(), HttpMethod.GET, request, String.class);
+            JSONObject jsonObject = new JSONObject(response.getBody().toString());
+            ProjectDetails projectDetails = getProjectDetails(jsonObject);
+            model.addAttribute("project", projectDetails);
+        } else {
+            model.addAttribute("project", new ProjectDetails());
+        }
+
+        return "admin_monthly_contribute";
+    }
+
+    @PostMapping("/admin/monthly/contribute")
+    public String adminMonthlyValidate(@Valid @ModelAttribute("project") ProjectDetails project,
+                                       BindingResult binding, HttpSession session, Model model) throws WebServiceRuntimeException {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        if (binding.hasErrors()) {
+            StringBuilder message = new StringBuilder();
+            message.append("Error(s):");
+            message.append("<ul class=\"fa-ul\">");
+            for (ObjectError objectError : binding.getAllErrors()) {
+                FieldError fieldError = (FieldError) objectError;
+                message.append("<li><i class=\"fa fa-exclamation-circle\"></i> ");
+                switch (fieldError.getField()) {
+                    case "organisationType":
+                        message.append("Organisation Type ");
+                        message.append(fieldError.getDefaultMessage());
+                        break;
+                    case "organisationName":
+                        message.append("Organisation Name ");
+                        message.append(fieldError.getDefaultMessage());
+                        break;
+                    case "projectName":
+                        message.append("Project Name ");
+                        message.append(fieldError.getDefaultMessage());
+                        break;
+                    case "owner":
+                        message.append("Owner ");
+                        message.append(fieldError.getDefaultMessage());
+                        break;
+                    case "dateCreated":
+                        message.append("Date Created ");
+                        message.append(fieldError.getDefaultMessage());
+                        break;
+                    default:
+                        message.append(fieldError.getField());
+                        message.append(" ");
+                        message.append(fieldError.getDefaultMessage());
+                }
+                message.append("</li>");
+            }
+            message.append("</ul>");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "</ul>" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)